lib/classes/eik-config.js

```diff
@@ -1,19 +1,54 @@
+// @ts-check
+/* eslint-disable no-continue */
 const { promisify } = require('util');
 const path = require('path');
+const { extname, join } = require('path');
+const isGlob = require('is-glob');
 const glob = promisify(require('glob'));
 const NoFilesMatchedError = require('./no-files-matched-error');
 const SingleDestMultipleSourcesError = require('./single-dest-multiple-source-error');
+const FileMapping = require('./file-mapping');
+const LocalFileLocation = require('./local-file-location');
+const RemoteFileLocation = require('./remote-file-location');
 const { assert, schema } = require('../schemas');
+const { typeSlug, pathDiff } = require('../helpers');
 
 const _config = Symbol('config');
 const _tokens = Symbol('tokens');
 
+/**
+ * @param {string} val
+ *
+ * @returns {string}
+ */
+const removeTrailingSlash = (val) =>
+    val.endsWith('/') ? val.substr(0, val.length - 1) : val;
+
+/**
+ * @param {string} val
+ *
+ * @returns {string}
+ */
+const addTrailingSlash = (val) =>
+    val.endsWith('/') ? val : `${val}/`;
+
```

Comment:
Are these methods intended to be used for when working on paths? Would using the node `join`  module where building paths solve this for instead of rolling our own? I mean, `join`  does normalize the slashes when joining paths.

Reply:
it doesn't normalise leading or trailing slashes at all unfortunately

Reply:
```
> join('asd', 'asd', 'asd')
'asd/asd/asd'
> join('/asd', 'asd', 'asd')
'/asd/asd/asd'
> join('/asd', 'asd', 'asd/')
'/asd/asd/asd/'
> 
```